app/js/components/LabResultsList.jsx

```diff
@@ -127,6 +131,31 @@ export class LabResultsList extends PureComponent {
     }
   }
 
+  filterData(filters, data, locale) {
+
+    let filteredData = data;
+
+    if (filters.dateField !== undefined && filters.dateField === "obsDatetime") {
+      if (filters.dateToField && filters.dateFromField) {
+        filteredData = getDateRange(filteredData, filters.dateFromField, filters.dateToField, filters.dateField);
+      }
+    }
+
+    if (filters.testTypeField !== undefined && filters.testTypeField !== "" && filters.testTypeField!== "All") {
+      const inputValue = filters.testTypeField;
+      filteredData = filteredData.filter((labTest) => {
+         return labTest.concept.uuid === inputValue
+           || (labTest.groupMembers && labTest.groupMembers.some((panelMember) => {
+              return panelMember.concept.uuid === inputValue
+        }))
```

Comment:
An object passes the filter if it is the selected lab test, or if it is a panel containing the selected lab test.